enterprise/server/workflow/service/service.go

```diff
@@ -1181,6 +1188,11 @@ func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tabl
 			},
 		},
 	}
+	// See https://github.com/buildbuddy-io/buildbuddy-internal/issues/3101
+	// for more info on why we don't want to pass empty flags
+	if len(string(envJson)) > 0 {
```

Comment:
need `len(env) > 0` here instead - if `len(env) == 0` then `envJson` will be `{}` I think.

Reply:
Ah good catch thanks